youtube_dl/YoutubeDL.py

```diff
@@ -563,7 +563,7 @@ def prepare_filename(self, info_dict):
             if autonumber_size is None:
                 autonumber_size = 5
             autonumber_templ = '%0' + str(autonumber_size) + 'd'
-            template_dict['autonumber'] = autonumber_templ % self._num_downloads
+            template_dict['autonumber'] = autonumber_templ % (self.params.get('autonumber_start') - 1 + self._num_downloads)
```

Comment:
You'll have to provide a default value for `autonumber_start` (like 1), because when using the youtube_dl module from python it will usually be missing.